### ISSUE TITLE
Error when installing perception_ros

### ISSUE THREAD

**Issue by yangrongkun** [hmz-15/Interactive-Scene-Reconstruction#13]
Dear authors, thank you for your great work.
When build with the catkin, there is an error occured for perception_ros, the error message is below, could you give some advice?

=======================================================
/src/Interactive-Scene-Reconstruction/mapping/perception_ros/perception_ros/src/pc_seg_generator_node.cpp:86:13: note: in expansion of macro ‘ROS_ERROR’
             ROS_ERROR("Mismatch between pano_seg_class and sem_class_with_color: ", category_name);
             ^~~~~~~~~
/home/yrk/Desktop/program/InteractiveCAD/src/Interactive-Scene-Reconstruction/mapping/perception_ros/perception_ros/src/pc_seg_generator_node.cpp: In member function ‘int PerceptionROS::PCSegGeneratorNode::GetMappedSemanticLabel(int)’:
/home/yrk/Desktop/program/InteractiveCAD/src/Interactive-Scene-Reconstruction/mapping/perception_ros/perception_ros/src/pc_seg_generator_node.cpp:939:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
/usr/bin/ld: warning: libopencv_imgcodecs.so.3.4, needed by /usr/local/lib/libopencv_highgui.so.3.4.15, may conflict with libopencv_imgcodecs.so.3.2
/usr/bin/ld: warning: libopencv_imgproc.so.3.4, needed by /usr/local/lib/libopencv_highgui.so.3.4.15, may conflict with libopencv_imgproc.so.3.2
CMakeFiles/pc_seg_generator_node.dir/src/pc_seg_generator_node.cpp.o: In function `PerceptionROS::PCSegGeneratorNode::ProcessImageMsg(boost::shared_ptr<sensor_msgs::Image_<std::allocator<void> > const> const&, boost::shared_ptr<sensor_msgs::Image_<std::allocator<void> > const> const&)':
pc_seg_generator_node.cpp:(.text+0x5658): undefined reference to `cv::rgbd::rescaleDepth(cv::_InputArray const&, int, cv::_OutputArray const&)'
CMakeFiles/pc_seg_generator_node.dir/src/pc_seg_generator_node.cpp.o: In function `PerceptionROS::PCSegGeneratorNode::~PCSegGeneratorNode()':
pc_seg_generator_node.cpp:(.text+0x12fac): undefined reference to `cv::rgbd::RgbdNormals::~RgbdNormals()'
/home/yrk/Desktop/program/InteractiveCAD/devel/lib/libperception_ros.so: undefined reference to `vtable for cv::rgbd::RgbdNormals'
/home/yrk/Desktop/program/InteractiveCAD/devel/lib/libperception_ros.so: undefined reference to `cv::rgbd::RgbdNormals::RgbdNormals(int, int, int, cv::_InputArray const&, int, int)'
/home/yrk/Desktop/program/InteractiveCAD/devel/lib/libperception_ros.so: undefined reference to `cv::rgbd::RgbdNormals::operator()(cv::_InputArray const&, cv::_OutputArray const&) const'
/home/yrk/Desktop/program/InteractiveCAD/devel/lib/libperception_ros.so: undefined reference to `cv::rgbd::depthTo3d(cv::_InputArray const&, cv::_InputArray const&, cv::_OutputArray const&, cv::_InputArray const&)'
collect2: error: ld returned 1 exit status
make[2]: *** [/home/yrk/Desktop/program/InteractiveCAD/devel/lib/perception_ros/pc_seg_generator_node] Error 1
make[1]: *** [CMakeFiles/pc_seg_generator_node.dir/all] Error 2
make: *** [all] Error 2

**Comment by yangrongkun** [comment]
My system is ubuntu 18.04.

**Comment by TooSchoolForCool** [comment]
There appears to be an issue with the OpenCV library. 

The version you're using doesn't seem to include the function `cv::rgbd::RgbdNormals::operator()(cv::_InputArray const&, cv::_OutputArray const&) const`. 

You may need to install a compatible version of OpenCV to resolve this problem.

**Comment by yangrongkun** [comment]
Yes, I guess this is the problem too, I use the opencv 3.4.15, can you share the detail version of opencv,
Thank you.

**Comment by yangrongkun** [comment]
When I read the opencv doc, 3.4.15 seems have the function cv::rgbd::RgbdNormals
